### ISSUE TITLE
"Backend not available"

### ISSUE THREAD

**Issue by Roi03** [retorquere/zotero-report-customizer#41]
When I attempt to save my changes to the report, I get an error: "backend not available."

**Comment by retorquere** [comment]
Are you using Juris-M or Zotero?

**Comment by Roi03** [comment]
Zotero

**Comment by retorquere** [comment]
And what operating system do you use?

**Comment by Roi03** [comment]
Win 8.1

**Comment by retorquere** [comment]
It's a little harder on Windows to do diagnosis. Can you go to 

`Help` -> `Debug Output Logging` -> `Restart with Logging Enabled`

let Zotero restart, then go into 

`Help` -> `Debug Output Logging` -> `View Output`, 

and in the window that pops up choose `File` -> `Save`.

You'll get a file picker popup, in that popup choose

`Format: Text Files`

and click `Save`. 

You now have a log file. It may have been named `Debug Output.html`, rename it to have a `.txt` extension.

Then finally, attach it here (**do not paste the contents** into a text field here, just drag the file on a comment field)

**Comment by Roi03** [comment]
[Debug Output.txt](https://github.com/retorquere/zotero-report-customizer/files/3958344/Debug.Output.txt)

**Comment by retorquere** [comment]
Can you do this again, but try to replicate the problem between `Restart with Logging Enabled` and `View Output`? (I should have said that, sorry)

**Comment by Roi03** [comment]
[Debug Output.txt](https://github.com/retorquere/zotero-report-customizer/files/3958486/Debug.Output.txt)

**Comment by small-fork** [comment]
Hello, I've had this very same issue occur. Wasn't sure if I should submit my own issue or add to this one? If it's preferred to add here, I've run through the steps you outlined above and am attaching my log file.

I'm running Zotero on a Mac with High Sierra version 10.13.6
[Debug Output.txt](https://github.com/retorquere/zotero-report-customizer/files/4049713/Debug.Output.txt)

**Comment by retorquere** [comment]
I'm trying to get it to work again but for some reason a ton of stuff isn't working. Maybe related with the Zotero upgrade a while ago since that switched to a newer Firefox base.

**Comment by retorquere** [comment]
:robot: this is your friendly neighborhood build bot announcing [test build 5.0.32.150](https://github.com/retorquere/zotero-report-customizer/releases/download/builds/zotero-report-customizer-5.0.32.150.xpi) ("icons load again")

Install in Zotero by downloading [test build 5.0.32.150](https://github.com/retorquere/zotero-report-customizer/releases/download/builds/zotero-report-customizer-5.0.32.150.xpi), opening the Zotero "Tools" menu, selecting "Add-ons", open the gear menu in the top right, and select "Install Add-on From File...".

**Comment by retorquere** [comment]
Does .150 work for you guys?

**Comment by small-fork** [comment]
@retorquere apologies for the delay. I installed the downloaded file, but am still getting the error message:
<img width="1068" alt="image" src="https://user-images.githubusercontent.com/27249070/72779988-21dd7d00-3be3-11ea-98e7-19f9e5beb6b8.png">

**Comment by kirk86** [comment]
I have the same issue, currently running 5.0.32

**Comment by retorquere** [comment]
I've found the problem, it will take a bit to get this fixed, but I'm working on it.

**Comment by retorquere** [comment]
:robot: this is your friendly neighborhood build bot announcing [test build 5.0.32.151](https://github.com/retorquere/zotero-report-customizer/releases/download/builds/zotero-report-customizer-5.0.32.151.xpi) ("this should do it")

Install in Zotero by downloading [test build 5.0.32.151](https://github.com/retorquere/zotero-report-customizer/releases/download/builds/zotero-report-customizer-5.0.32.151.xpi), opening the Zotero "Tools" menu, selecting "Add-ons", open the gear menu in the top right, and select "Install Add-on From File...".

**Comment by small-fork** [comment]
Hi @retorquere Thank you for the update. I'm finding now that while I'm not getting an error upon clicking the "save" icon, I can't tell if it's meant to do something upon the "Save" click? Or does one click that and then use the Toolbar File > Save from that point?

Thank you again for patching this up!

**Comment by retorquere** [comment]
"save" saves the current setup - a save to file won't pick up the changes until you've done that. I can't fix the latter, but if the save icon is disabled, it's safe to save to disk.

**Comment by retorquere** [comment]
:robot: this is your friendly neighborhood build bot announcing [test build 5.0.32.152](https://github.com/retorquere/zotero-report-customizer/releases/download/builds/zotero-report-customizer-5.0.32.152.xpi) ("cleanup")

Install in Zotero by downloading [test build 5.0.32.152](https://github.com/retorquere/zotero-report-customizer/releases/download/builds/zotero-report-customizer-5.0.32.152.xpi), opening the Zotero "Tools" menu, selecting "Add-ons", open the gear menu in the top right, and select "Install Add-on From File...".

**Comment by retorquere** [comment]
:robot: this is your friendly neighborhood build bot announcing [test build 5.0.32.153](https://github.com/retorquere/zotero-report-customizer/releases/download/builds/zotero-report-customizer-5.0.32.153.xpi) ("cleanup")

Install in Zotero by downloading [test build 5.0.32.153](https://github.com/retorquere/zotero-report-customizer/releases/download/builds/zotero-report-customizer-5.0.32.153.xpi), opening the Zotero "Tools" menu, selecting "Add-ons", open the gear menu in the top right, and select "Install Add-on From File...".

**Comment by Roi03** [comment]
It work's perfectly now, thanks! Dumb question is there an option to add to the report the file location (C:\) instead of the "Select in Zotero" option? Only reason is to access the document without opening the program. Cheers.

**Comment by retorquere** [comment]
I don't understand what you're asking, sorry. Can you rephrase?

**Comment by Roi03** [comment]
I mean return the attachment location in the disk, so as to open the PDF/HTML without opening the main zotero program.

**Comment by retorquere** [comment]
Ah, I see. I can look into that, but that's a separate feature, so I'd appreciate it if you could open a new issue.
